static/app/components/smartSearchBar/index.tsx

```diff
@@ -464,6 +464,22 @@ class SmartSearchBar extends React.Component<Props, State> {
     callIfFunction(this.props.onChange, evt.target.value, evt);
   };
 
+  /**
+   * Prevent pasting extra spaces from formatted text
+   */
+  onPaste = (evt: React.ClipboardEvent<HTMLTextAreaElement>) => {
+    // cancel paste
+    evt.preventDefault();
+
+    // get text representation of clipboard
+    const text = evt.clipboardData.getData('text/plain').replace('\n', '').trim();
+
+    // insert text manually
+    const query = this.state.query + text;
+    this.setState(makeQueryState(query), this.updateAutoCompleteItems);
+    callIfFunction(this.props.onChange, query, evt);
+  };
```

Comment:
What happens when I paste at the front, or middle?

Reply:
we fixed this